### PR TITLE
Meta: Infer better types in `querySelector` and `select-dom`

### PR DESCRIPTION
**Pull request by fregante**
Testing out https://github.com/g-plane/typed-query-selector with https://github.com/fregante/select-dom/pull/15, but unfortunately the types currently have some issues.

### CODE REVIEW

**Review by sugam74 — APPROVED** [review]

**Comment by fregante** [comment]
Still blocked by:

- [ ] https://github.com/g-plane/typed-query-selector/pull/7
- [ ] https://github.com/fregante/select-dom/pull/15#issuecomment-751433933

**Comment by fregante** [comment]
Changes so far have been double-checked and should be correct.

File: source/github-helpers/index.ts [hunk 1]
```diff
@@ -52,11 +52,12 @@ export const buildRepoURL = (...pathParts: Array<string | number> & {0: string})
 };
 
 export const getPRHeadRepo = (): ReturnType<typeof getRepo> => {
-	return getRepo(select<HTMLAnchorElement>('.commit-ref.head-ref a')!);
+	const headLink = select('.commit-ref.head-ref a');
+	return getRepo(headLink);
```
Comment (fregante) [inline source/github-helpers/index.ts:56]: XO was complaining about the ! on GHA, but TypeScript wouldn't accept it without. This way it works for both. Don't ask me why

**Comment by fregante** [comment]
Changes so far have been double-checked and should be correct.

**Comment by fregante** [comment]
I'll merge this _after_ the next version, so we test out the new changes in `element-ready` first

**Comment by yakov116** [comment]
> I'll merge this _after_ the next version, so we test out the new changes in `element-ready` first

Heads up you turned off the auto release

**Comment by fregante** [comment]
Yep, I want v21.1.1 😃

**Comment by yakov116** [comment]
🎉

File: source/features/conversation-filters.tsx [hunk 2]
```diff
@@ -6,15 +6,11 @@ import SearchQuery from '../github-helpers/search-query';
 import {getUsername} from '../github-helpers';
 
 function init(): void {
-	// Use an existing dropdown item to preserve its DOM structure (supports old GHE versions)
-	const sourceItem = select<HTMLAnchorElement>([
-		'#filters-select-menu a:nth-last-child(2)', // GHE
```
Comment (yakov116) [inline source/features/conversation-filters.tsx:11]: Looks like this was wrongly labeled ⬆️ works ⬇️ does not

Comment (yakov116) [reply source/features/conversation-filters.tsx:11]: Was added [Here](https://github.com/sindresorhus/refined-github/pull/2809/files)

**Comment by fregante** [comment]
> Very nice work!

Thank @g-plane!

**Pull request merged**